### ISSUE TITLE
sc.pl.highest_expr_genes() with layer errors

### ISSUE THREAD

**Issue by mezwick** [scverse/scanpy#3318]
### Please make sure these conditions are met

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of scanpy.
- [x] (optional) I have confirmed this bug exists on the main branch of scanpy.

### What happened?

Trying to store normalised values in a layer 'normalised', then plot from that layer with sc.pl.highest_expr_genes(). But the function fails with the layer parameter.

### Minimal code sample

```py
import numpy as np
import pandas as pd
import anndata as ad

# Create a small data matrix
data = np.random.rand(10, 5)

# Create observation (cell) and variable (gene) annotations
obs = pd.DataFrame(index=[f'Cell_{i}' for i in range(data.shape[0])])
var = pd.DataFrame(index=[f'Gene_{i}' for i in range(data.shape[1])])

# Create the AnnData object
adata = ad.AnnData(X=data, obs=obs, var=var)

# Test layer call function
adata.layers['normalised'] = adata.X # not doing any manipulation of X at the moment to test
sc.pl.highest_expr_genes(adata, layer='normalised')
```

### Error output

```pytb
Output exceeds the size limit. Open the full output data in a text editor
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[32], line 17
     15 # Test layer call function
     16 adata.layers['normalised'] = adata.X # not doing any manipulation of X at the moment to test
---> 17 sc.pl.highest_expr_genes(adata, layer='normalised')
     19 # Test layer call function
     20 adata.layers['normalised'] = adata.X # not doing any manipulation of X at the moment to test

File c:\Users\U062951\Miniconda3\envs\imc_ome_v1_spatial\lib\site-packages\legacy_api_wrap\__init__.py:80, in legacy_api.<locals>.wrapper.<locals>.fn_compatible(*args_all, **kw)
     77 @wraps(fn)
     78 def fn_compatible(*args_all: P.args, **kw: P.kwargs) -> R:
     79     if len(args_all) <= n_positional:
---> 80         return fn(*args_all, **kw)
     82     args_pos: P.args
     83     args_pos, args_rest = args_all[:n_positional], args_all[n_positional:]

File c:\Users\U062951\Miniconda3\envs\imc_ome_v1_spatial\lib\site-packages\scanpy\plotting\_qc.py:100, in highest_expr_genes(adata, n_top, show, save, ax, gene_symbols, log, **kwds)
     98     height = (n_top * 0.2) + 1.5
     99     fig, ax = plt.subplots(figsize=(5, height))
--> 100 sns.boxplot(data=counts_top_genes, orient="h", ax=ax, fliersize=1, **kwds)
    101 ax.set_xlabel("% of total counts")
    102 if log:

File c:\Users\U062951\Miniconda3\envs\imc_ome_v1_spatial\lib\site-packages\seaborn\categorical.py:1634, in boxplot(data, x, y, hue, order, hue_order, orient, color, palette, saturation, fill, dodge, width, gap, whis, linecolor, linewidth, fliersize, hue_norm, native_scale, log_scale, formatter, legend, ax, **kwargs)
...
--> 700 artists = ax.bxp(**boxplot_kws)
    702 # Reset artist widths after adding so everything stays positive
    703 ori_idx = ["x", "y"].index(self.orient)

TypeError: Axes.bxp() got an unexpected keyword argument 'layer'
```

### Versions
<details>

```
scanpy -v 1.10.3
anndata -v 0.10.9
python -v 3.10
```

</details>

**Comment by flying-sheep** [comment]
Well, the documentation of `highest_expr_genes` doesn’t say that it supports layers, but this is a very sensible feature request.

**Comment by mezwick** [comment]
sorry, copilot hallucinated me a layer variable!

**Comment by flying-sheep** [comment]
Ha, it has a point. I’m going to reopen this, since I already fixed this in #3324
